.github/workflows/docs.yml

```diff
@@ -8,11 +8,17 @@ on:
       - main
       - test-docs-generator # for testing
 
+permissions:
+  contents: read
+
 jobs:
   update:
     name: Update
     if: github.event.pull_request.merged == true
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
```

Comment:
With this we could define top level as
```
permissions: {}
```
...or remove this line as it's currently redundant with toplevel.